src/compiler/moduleNameResolver.ts

```diff
@@ -2392,6 +2396,44 @@ function getVersionPathsOfPackageJsonInfo(packageJsonInfo: PackageJsonInfo, stat
     return packageJsonInfo.contents.versionPaths || undefined;
 }
 
+function getPeerDependenciesOfPackageJsonInfo(packageJsonInfo: PackageJsonInfo, state: ModuleResolutionState): string | undefined {
+    if (packageJsonInfo.contents.peerDependencies === undefined) {
+        packageJsonInfo.contents.peerDependencies = readPackageJsonPeerDependencies(packageJsonInfo, state) || false;
+    }
+    return packageJsonInfo.contents.peerDependencies || undefined;
+}
+
+function readPackageJsonPeerDependencies(packageJsonInfo: PackageJsonInfo, state: ModuleResolutionState): string | undefined {
+    const peerDependencies = readPackageJsonField(packageJsonInfo.contents.packageJsonContent, "peerDependencies", "object", state);
+    if (peerDependencies === undefined) return undefined;
+    if (state.traceEnabled) trace(state.host, Diagnostics.package_json_has_a_peerDependencies_field);
+    const packageDirectory = realPath(packageJsonInfo.packageDirectory, state.host, state.traceEnabled);
+    if (packageDirectory.indexOf(".pnpm") === -1) {
+        // For now skip extra work for anything else
+        if (state.traceEnabled) trace(state.host, Diagnostics.Package_is_not_installed_using_pnpm_skipping_peerDependencies_version_resolution);
+        return undefined;
+    }
+    const nodeModules = packageDirectory.substring(0, packageDirectory.lastIndexOf("node_modules") + "node_modules".length) + directorySeparator;
+    let result = "";
+    if (state.traceEnabled) {

```

Comment:
If trace isn't enabled, won't this do nothing and return the empty string?

Reply:
oops

Reply:
No tests changed; were they generated before this code was added and so the fix is still workign?

Reply:
This only takes place if it’s at .pnpm location and has peer dependencies.. as far as I know it’s the only test I have and it was with trace resolutions so didn’t change any baseline

Reply:
Ah, tracing was enabled, right.